chromium/browser_type.go

```diff
@@ -92,7 +92,7 @@ func (b *BrowserType) init(
 }
 
 func (b *BrowserType) initContext() context.Context {
-	ctx := k6ext.WithVU(b.vu.Context(), b.vu)
+	ctx := k6ext.WithVU(context.Background(), b.vu)
```

Comment:
Can we initialize this context before calling `initContext` and then calling `initContext(ctx)`? So that we can have an explicit lifecycle, manage it from tests, etc.

Reply:
Partially related with this comment and [this comment](https://github.com/grafana/xk6-browser/pull/910#discussion_r1210096528), I remember experimenting with passing `context` to `Launch` and `Connect` methods unsuccessfully, because the calling point was other and `ctx` was being closed, but now, calling from the mapped methods this is working, which improves our implementation and allows us to get rid of https://github.com/grafana/xk6-browser/pull/910/commits/b08de580ac67cb56d0ed4634f7c21647a160e007. See https://github.com/grafana/xk6-browser/pull/910/commits/a23399ed896732557fb0a4ced33c46754739b039.
Thanks for bringing this up!

Reply:
Nice, thanks! Should we also pass the context to `getOrInitBrowser` and leave the context initialization to `getOrInitBrowser` for the same reason?

Reply:
Yep, I did this kind of naively, my bad, it makes sense. See [4464d33](https://github.com/grafana/xk6-browser/pull/910/commits/4464d33337ac1ff9f65144449f0b53d88f9b233a).